### PR TITLE
ssh plugin support specifying private/public keys path

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Added `ssh-key-file-path` to allow users to specify the path of ssh keys. For example if you want to store the keys in `/a/b`, the blowing config can work.

```
apiVersion: batch.volcano.sh/v1alpha1
kind: Job
metadata:
  name: lm-horovod-job
  labels:
    "volcano.sh/job-type": Horovod
spec:
  minAvailable: 4
  schedulerName: volcano
  plugins:
    ssh: ["--ssh-key-file-path=/a/b"]    # or ["--ssh-key-file-path", "/a/b"]
```

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
Fixes: https://github.com/volcano-sh/volcano/issues/403

File: pkg/controllers/job/plugins/ssh/ssh.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	"encoding/pem"
 	"flag"
 	"fmt"
+	"volcano.sh/volcano/pkg/controllers/job/plugins/env"
```
Comment (k82cn) [inline pkg/controllers/job/plugins/ssh/ssh.go]: follow import order

File: pkg/controllers/job/plugins/ssh/ssh.go [hunk 2]
```diff
@@ -187,7 +191,10 @@ func (sp *sshPlugin) cmName(job *batch.Job) string {
 
 func (sp *sshPlugin) addFlags() {
 	flagSet := flag.NewFlagSet(sp.Name(), flag.ContinueOnError)
+	// TODO: deprecate no-root
 	flagSet.BoolVar(&sp.noRoot, "no-root", sp.noRoot, "The ssh user, --no-root is common user")
+	flagSet.StringVar(&sp.sshKeyFilePath, "ssh-key-file-path", sp.sshKeyFilePath, "The path used to store "+
+		"ssh private and public keys, by default it is `/root/.ssh`")
```
Comment (k82cn) [inline pkg/controllers/job/plugins/ssh/ssh.go]: it is `/root/.ssh` by default

**Comment by k82cn** [comment]
please list the value table for `--no-root` and `ssh-key-file-path`.

File: pkg/controllers/job/plugins/ssh/ssh.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/controllers/job/plugins/ssh/ssh.go]: ok, seems we should add a ci for this.

**Comment by hzxuzhonghu** [comment]
>please list the value table for --no-root and ssh-key-file-path.

Sorry, i can not catch this.

**Comment by hzxuzhonghu** [comment]
ping @k82cn

**Comment by dalfos** [comment]
@hzxuzhonghu have you example of job how to use ssh & mpi in userspace?

**Comment by k82cn** [comment]
@hzxuzhonghu , please help to list expected result when `--no-root` & `--ssh-key` have different value.

**Comment by k82cn** [comment]
> @hzxuzhonghu have you example of job how to use ssh & mpi in userspace?

@dalfos , a initContainer or script is necessary to copy private/public key into `~/.ssh` to run mpi job in userspace.

**Comment by dalfos** [comment]
@k82cn maybe add param --port $port which adds lines "Port $port" for master and workers in .ssh/config. User cannot start sshd on port 22 (I started for example on port 2222).

**Comment by dalfos** [comment]
> @k82cn maybe add param --port $port which adds lines "Port $port" for master and workers in .ssh/config. User cannot start sshd on port 22 (I started for example on port 2222).

I found in mpirun:
 -mca plm_rsh_args "-p 2222"

**Comment by hzxuzhonghu** [comment]
Yes, we donot care about the port of sshd in volcano.

>@hzxuzhonghu have you example of job how to use ssh & mpi in userspace?

What do u mean by userspace?


>@hzxuzhonghu , please help to list expected result when --no-root & --ssh-key have different value.

if `--ssh-key-file-path`  is specified, `no-root` is not cared about, i want to remove the `no-root` flag in future.

**Comment by dalfos** [comment]
@hzxuzhonghu I mean case then sshd is executed by user (not by root).
All my questions are settled. Thank you!

**Comment by k82cn** [comment]
> if --ssh-key-file-path is specified, no-root is not cared about, i want to remove the no-root flag in future.

ok, so the default value of `--ssh-key-file-path` will be `/root/.ssh` ?

**Comment by hzxuzhonghu** [comment]
Yes

**Pull request merged**